### PR TITLE
[WIP]💈 Bare bert config: no engine, no evaluation.

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
1. Bare bert config: no engine, no evaluation. https://github.com/microsoft/Olive/pull/1013
2. Fix a few bugs:
 - self._user_module_loader is not None even user_script, dataloader_func and etc. are not given.
 - inc quantization's calibration dataloader input requires (input, label) which is data_config's data_loader.
   - https://github.com/microsoft/Olive/pull/1012

Just send this PR to clarify bare transformer olive config user need to give:
1. when enable auto_optimizer, user needs provide input model, data_config and auto_optimizer_config at least.
2. when disable auto_optimizer, user needs provide input_model, passes at least.

For the bugfix, will try to send the separated PR with UT one-by-one.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: examples/bert/bert_auto_opt_gpu copy.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (xiaoyu-work) [inline examples/bert/bert_auto_opt_gpu copy.json:1]: typo: remove `copy` from the file name.

Comment (trajepl) [reply examples/bert/bert_auto_opt_gpu copy.json:1]: Thanks~ I just  send this PR to clarify bare transformer olive config user need to give.  There are separated PRs to achieve this `bare` json.
Please refer to description of this PR.

Comment (trajepl) [reply examples/bert/bert_auto_opt_gpu copy.json:1]: Will close this once the separated PRs are done.

File: olive/auto_optimizer/regulate_mixins.py [hunk 2]
```diff
@@ -37,6 +37,10 @@ def regulate_pass_flows_dict(self, pass_flows_dict):
                     if p not in pass_config:
                         pass_config.update({p: {"type": p, "config": {}}})
 
+        # disable pass search when search strategy is None/False
```
[the remark this answers is not in the document]
Comment (trajepl) [reply olive/auto_optimizer/regulate_mixins.py:40]: image

search strategy is invisible to auto optimizer.

**Comment by trajepl** [comment]
Closed this PR as the required changes already be done.

**Pull request closed without merging**